### PR TITLE
Governance: Extend Veto vote period by min_transaction_hold_up_time

### PR DESCRIPTION
**Pull request by SebastianBor**

### CODE REVIEW

**Comment by SebastianBor** [comment]
The build issue is fixed by https://github.com/solana-labs/solana-program-library/pull/3818

File: governance/program/src/state/proposal.rs [hunk 1]
```diff
@@ -260,23 +260,88 @@ impl ProposalV2 {
...
+        current_unix_timestamp: UnixTimestamp,
+    ) -> Result<(), ProgramError> {
+        match self.state {
+            // Veto vote can be cast when:
+            // Proposal is in Voting state and within max_voting_time + min_transaction_hold_up_time period
+            ProposalState::Voting => {
+                if self
+                    .vote_end_time(governance_config)
+                    .checked_add(governance_config.min_transaction_hold_up_time as i64)
+                    .unwrap()
```
Comment (joncinque) [inline governance/program/src/state/proposal.rs:312]: nit: a governance with a huge `min_transaction_hold_up_time` is essentially useless, but if they happen to wrongly configure it this way, you can avoid a panic here with a `saturating_add`

File: governance/program/src/state/proposal.rs [hunk 2]
```diff
@@ -260,23 +260,88 @@ impl ProposalV2 {
...
+                    Ok(())
+                }
+            }
+            // Proposal is in Succeeded state and within min_transaction_hold_up_time period calculated from the time when the vote ended (voting_completed_at)
+            ProposalState::Succeeded => {
+                if self
+                    .voting_completed_at
+                    .unwrap()
+                    .checked_add(governance_config.min_transaction_hold_up_time as i64)
+                    .unwrap()
```
Comment (joncinque) [inline governance/program/src/state/proposal.rs:326]: nit: same here

**Review by joncinque — APPROVED** [review]
Looks great!

File: governance/program/src/state/proposal.rs (hunk 1 above)
Comment (SebastianBor) [reply governance/program/src/state/proposal.rs]: done

File: governance/program/src/state/proposal.rs (hunk 2 above)
Comment (SebastianBor) [reply governance/program/src/state/proposal.rs]: done

**Pull request merged**